### PR TITLE
datolist

### PR DESCRIPTION
**Pull request by theblushingcrow**

### CODE REVIEW

File: lib/src/accordion/accordion-header/accordion-header.component.ts [hunk 1]
```diff
@@ -6,17 +6,18 @@
...
 
-import { Component, ElementRef, HostBinding, Input, ChangeDetectorRef } from '@angular/core';
+import { Component, ElementRef, HostBinding, Input, ChangeDetectorRef, ChangeDetectionStrategy } from '@angular/core';
 import { fromEvent, Observable } from 'rxjs';
 
 @Component({
   selector: 'dato-accordion-header',
-  template: '<ng-content></ng-content>',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: '<div class="d-flex items-center">' + '<dato-icon *ngIf="includeArrow && !_expanded" datoIcon="arrow-right" datoSize="sm" class="mx-8"></dato-icon>' + '<dato-icon *ngIf="includeArrow && _expanded" datoIcon="arrow-down" datoSize="sm" class="mx-8"></dato-icon>' + '<ng-content></ng-content></div>',
```
Comment (NetanelBasal) [inline lib/src/accordion/accordion-header/accordion-header.component.ts]: move to HTML file.

File: lib/src/list/list.component.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
+$md: (
+        height: 35px,
+        selectedHeight: 25px,
+        optionHeight: 35px,
+        selectedMargin: 5px,
+        font: 14px
+);
+
+dato-list {
+  & .dato-list__options{
```
Comment (NetanelBasal) [inline lib/src/list/list.component.scss]: remove

File: lib/src/list/list.component.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    height: map-get($md, optionHeight);
+    line-height: map-get($md, optionHeight);
+    font-size: map-get($md, font);
+  }
+  & dato-accordion .dato-select__option {
+    margin-left: 15px;
+  }
+}
+
+.force-hide {
```
Comment (NetanelBasal) [inline lib/src/list/list.component.scss]: should be in atomic

File: lib/src/list/list.component.ts [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+
+  /** Keyboard subscription */
+  private keyboardEventsManagerSubscription;
+
+  /** Search control subscription */
+  private searchSubscription;
+
+  constructor(private cdr: ChangeDetectorRef, private translate: DatoTranslateService, private host: ElementRef, @Attribute('datoSize') public size) {
+    super();
+    // this.size = size || 'md';
```
Comment (NetanelBasal) [inline lib/src/list/list.component.ts]: remove

File: lib/src/list/list.component.ts [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+   * Search for options and returns if we have result
+   * @param {string} value
+   * @returns {boolean}
+   */
+  private searchOptions(value: string) {
+    const results = [];
+    const groupComponentsArray = this.accordion.length ? this.accordion.first.groups.toArray() : this.groups.toArray();
+
+    this._data.forEach((group, index) => {
+      const matchGroup = group[this.labelKey].toLowerCase().indexOf(value) > -1;
```
Comment (NetanelBasal) [inline lib/src/list/list.component.ts]: You should support custom search strategy and don't limit to toLowerCase() search. Check the select.

File: lib/src/list/list.types.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/**
+ * @license
+ * Copyright Datorama. All Rights Reserved.
+ *
+ * Use of this source code is governed by an Apache License 2.0 license that can be
+ * found in the LICENSE file at https://github.com/datorama/client-core/blob/master/LICENSE
+ */
+
+export type ListSize = 'sm' | 'md' | 'lg';
```
Comment (NetanelBasal) [inline lib/src/list/list.types.ts]: Do we support all of them?

File: lib/src/options/group.component.ts [hunk 7]
```diff
@@ -6,19 +6,23 @@
...
 @Component({
   selector: 'dato-group',
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: `
-    <div class="dato-select__option dato-select__group">
-      <ng-content select="[groupLabel]"></ng-content>
-    </div>
-    <ng-content></ng-content>
-  `
+        <div class="dato-select__option dato-select__group">
```
Comment (NetanelBasal) [inline lib/src/options/group.component.ts]: move to HTML file

File: lib/src/list/list.types.ts (hunk 6 above)
Comment (theblushingcrow) [reply lib/src/list/list.types.ts]: nope, not sure this is needed. will remove for now.

File: lib/src/accordion/accordion-header/accordion-header.component.ts (hunk 1 above)
Comment (theblushingcrow) [reply lib/src/accordion/accordion-header/accordion-header.component.ts]: Done

File: lib/src/list/list.component.scss (hunk 2 above)
Comment (theblushingcrow) [reply lib/src/list/list.component.scss]: Done

File: lib/src/list/list.component.ts (hunk 4 above)
Comment (theblushingcrow) [reply lib/src/list/list.component.ts]: Done.

File: lib/src/list/list.component.ts (hunk 5 above)
Comment (theblushingcrow) [reply lib/src/list/list.component.ts]: Done.

**Pull request merged**